lib/ably/realtime/connection.rb

```diff
@@ -440,9 +439,11 @@ def create_websocket_transport
                 logger.debug { "Resuming connection key #{key} with serial #{serial}" }
               elsif connection_recoverable?
                 url_params.merge! recover: connection_recover_parts[:recover], connectionSerial: connection_recover_parts[:connection_serial]
+                recovered_msg_serial = connection_recover_parts[:msg_serial].to_i
                 logger.debug { "Recovering connection with key #{client.recover}" }
                 unsafe_once(:connected, :closed, :failed) do
                   client.disable_automatic_connection_recovery
+                  @client_msg_serial = recovered_msg_serial
```

Comment:
> not obvious by inspection whether this will run before or after connectionmanager.rb:103 (which resets the connectionSerial on a failed resume). 

Yup, it's why I have tests :)